src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs

```diff
@@ -35,9 +35,9 @@ internal AsymmetricAdapter(SecurityKey key, string algorithm, bool requirePrivat
         {
         }
 
-        internal AsymmetricAdapter(SecurityKey key, string algorithm, HashAlgorithm hashAlgorithm, bool requirePrivateKey)
+        internal AsymmetricAdapter(SecurityKey key, string algorithm, HashAlgorithm? hashAlgorithm, bool requirePrivateKey)
         {
-            HashAlgorithm = hashAlgorithm;
+            HashAlgorithm = hashAlgorithm!;
```

Comment:
I'm confused, the parameter explicitly allows for null, how can we say it is not null here?

Reply:
The property needs to be null:

`private HashAlgorithm? HashAlgorithm { get; set; }`

Reply:
See lines 26-30, the delegate used by the public method are set correctly depending on the context.